### DISCUSSION TITLE
Диагностика CommentedCodeDiagnostic.

### DISCUSSION THREAD

**Discussion by Denis-Miheykin** [1c-syntax/bsl-language-server#3340] [Ideas]
Добрый день.

Можно ли доработать диагностику на игнорирование многострочных комментариев с указанием произвольной метки в первой строке?

**Пример.**
Когда совместимость конфигурации позволяет использовать только аннотацию `Вместо` в расширениях, то чтобы не удалять типовой код, его оборачивают в блоки с метками, а ниже пишется свой код, чтобы при последующем обновлении было понятно, где чьи строки. Например, вот на скрине первая строка в многострочном комментарии `--ТиповойКод`:

![image](https://github.com/user-attachments/assets/c22357af-6d56-404b-8e67-27fc1049599b)

При текущей реализации блок будет подсвечен весь. Хотелось бы исключить многострочный комментарий из диагностики, если вначале есть служебная строка.

Как я понял из реализации диагностики, сейчас определяются группы комментариев, для каждой группы определяется первая и последняя строка и каждая группа проверяется построчно, если в строке есть исключающие префиксы(из параметра диагностики), то строка пропускается, но если хоть в одной строке нету префикса, то вся группа будет добавлена в хранилище диагностики с первой по последнюю строку. При таком подходе, чтобы исключить весь блок, нужно исключающий префикс добавить в каждую строку, что очень не удобно:

![image](https://github.com/user-attachments/assets/2be210a8-a5fe-4c5b-bb03-34b3f64e3e53)

**Comment by nixel2007** [comment]
Интересное пожелание, спасибо. Надо подумать, как это можно реализовать, либо исключать весь блок целиком либо вводить расчёт "скобок".

Как временное решение могу предложить использовать экранирующие комментарии, отключающие срабатывание диагностики целиком.

**Reply by Denis-Miheykin** [reply]
> Надо подумать, как это можно реализовать, либо исключать весь блок целиком либо вводить расчёт "скобок".

Ну вот весь блок целиком исключить было бы удобно.

> Как временное решение могу предложить использовать экранирующие комментарии, отключающие срабатывание диагностики целиком.

Да, была такая мысль тоже)
